README.md
<!--
This README describes the package. If you publish this package to pub.dev,
this README's contents appear on the landing page for your package.

For information about how to write a good package README, see the guide for
[writing package pages](https://dart.dev/guides/libraries/writing-package-pages).

For general information about developing packages, see the Dart guide for
[creating packages](https://dart.dev/guides/libraries/create-library-packages)
and the Flutter guide for
[developing packages and plugins](https://flutter.dev/developing-packages).
-->

Hey Folks,
Excited to share my first ever contribution to the community! To make your dashboards more livelier using our animated widget.Our animated backgrounds are created by experienced designers and animators, ensuring that each animation is both visually impressive and technically sound. 

## Features


https://user-images.githubusercontent.com/56824729/227963316-86b862e6-31cb-4f42-91f4-a86bd3eedbf0.mp4

![demo_home](https://user-images.githubusercontent.com/56824729/229801694-29860b58-2a2d-4570-b23b-8d97b5ed0181.gif)




## Getting started

In your pubspec.yaml:

```dart
dependencies:
  animated_hover_background: any
```

## Usage

```dart
AnimatedHoverBackground(YourWidget());
```

## Additional information

Our animated backgrounds are created by experienced designers and animators, ensuring that each animation is both visually impressive and technically sound.So why settle for static backgrounds when you can add a touch of magic to your projects with our animated backgrounds? Download our package today and take your projects to the next level!


CHANGELOG.md
## 0.0.2

bug fixes


test/animated_hover_background_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:animated_hover_background/src/animated_hover_background.dart';

void main() {
  test('adds one to input values', () {

  });
}


example/lib/main.dart
import 'dart:ui';

import 'package:animated_hover_background/animated_hover_background.dart';
import 'package:background_anim/custom_calender.dart';
import 'package:background_anim/home_view.dart';
import 'package:flutter/gestures.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
        title: 'Flutter Demo',
        debugShowCheckedModeBanner: false,
        theme: ThemeData(
          primarySwatch: Colors.blue,
        ),
        home: const HomeScreen());
  }
}

class HomeScreen extends StatefulWidget {
  const HomeScreen({super.key});

  @override
  State<HomeScreen> createState() => _HomeScreenState();
}

class _HomeScreenState extends State<HomeScreen> {
  String pageName = 'home';

  @override
  Widget build(BuildContext context) {
    return AnimatedHoverBackground(Scaffold(
      backgroundColor: Colors.transparent,
      body: pageName == "home" ? Home_Dash_Screen() : CustomCalender(),
      bottomNavigationBar: Stack(
        children: [
          Container(
            height: 50,
            width: MediaQuery.of(context).size.width,
            margin: const EdgeInsets.only(bottom: 20, right: 12, left: 12),
            decoration: BoxDecoration(
                borderRadius: BorderRadius.circular(20),
                color: Colors.white24,
                border: Border.all(color: Colors.grey.shade200)),
            child: Row(
              mainAxisAlignment: MainAxisAlignment.spaceAround,
              children: [
                GestureDetector(
                    onTap: () {
                      pageName = 'home';
                      setState(() {});
                    },
                    child: const Icon(Icons.dashboard_customize_outlined)),
                GestureDetector(
                    onTap: () {
                      pageName = 'calender';
                      setState(() {});
                    },
                    child: const Icon(Icons.calendar_month)),
              ],
            ),
          ),
        ],
      ),
    ));
  }
}


example/lib/custom_calender.dart
// ignore_for_file: prefer_final_fields

import 'package:flutter/material.dart';
import 'package:flutter_calendar_carousel/classes/event.dart';
import 'package:flutter_calendar_carousel/flutter_calendar_carousel.dart';
import 'package:intl/intl.dart';

class CustomCalender extends StatefulWidget {
  const CustomCalender({super.key});

  @override
  _CustomCalenderState createState() => _CustomCalenderState();
}

class _CustomCalenderState extends State<CustomCalender> {
  DateTime _currentDate = DateTime.now();
  DateTime _currentDate2 = DateTime.now();
  String _currentMonth = DateFormat.yMMM().format(DateTime.now());
  DateTime _targetDateTime = DateTime.now();

  late CalendarCarousel _calendarCarouselNoHeader;

  ScrollController _scrollController = ScrollController();

  static Widget _eventIcon = Container(
    decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: const BorderRadius.all(Radius.circular(1000)),
        border: Border.all(color: Colors.blue, width: 2.0)),
    child: const Icon(
      Icons.person,
      color: Colors.amber,
    ),
  );

  EventList<Event> _markedDateMap = EventList<Event>(
    events: {
      DateTime(2020, 2, 10): [
        Event(
          date: DateTime(2020, 2, 14),
          title: 'Event 1',
          icon: _eventIcon,
          dot: Container(
            margin: const EdgeInsets.symmetric(horizontal: 1.0),
            color: Colors.red,
            height: 5.0,
            width: 5.0,
          ),
        ),
        Event(
          date: DateTime(2020, 2, 10),
          title: 'Event 2',
          icon: _eventIcon,
        ),
        Event(
          date: DateTime(2020, 2, 15),
          title: 'Event 3',
          icon: _eventIcon,
        ),
      ],
    },
  );

  @override
  void initState() {
    _markedDateMap.add(
        DateTime(2023, 2, 25),
        Event(
          date: DateTime(2020, 2, 25),
          title: 'Event 5',
          icon: _eventIcon,
        ));

    _markedDateMap.add(
        DateTime(2020, 2, 10),
        Event(
          date: DateTime(2020, 2, 10),
          title: 'Event 4',
          icon: _eventIcon,
        ));

    _markedDateMap.addAll(DateTime(2019, 2, 11), [
      Event(
        date: DateTime(2019, 2, 11),
        title: 'Event 1',
        icon: _eventIcon,
      ),
      Event(
        date: DateTime(2019, 2, 11),
        title: 'Event 2',
        icon: _eventIcon,
      ),
      Event(
        date: DateTime(2019, 2, 11),
        title: 'Event 3',
        icon: _eventIcon,
      ),
    ]);
    _calendarCarouselNoHeader = CalendarCarousel<Event>(
      todayBorderColor: Colors.green,
      onDayPressed: (DateTime date, List<Event> events) {
        setState(() => _currentDate2 = date);
      },
      daysHaveCircularBorder: false,
      showOnlyCurrentMonthDate: false,
      weekendTextStyle: const TextStyle(
        color: Colors.red,
      ),
      thisMonthDayBorderColor: Colors.grey,
      weekFormat: false,
//      firstDayOfWeek: 4,
      markedDatesMap: _markedDateMap,
      height: 300.0,
      selectedDateTime: _currentDate2,
      targetDateTime: _targetDateTime,
      customGridViewPhysics: const NeverScrollableScrollPhysics(),
      markedDateCustomShapeBorder:
          const CircleBorder(side: BorderSide(color: Colors.transparent)),
      // ignore: prefer_const_constructors
      markedDateCustomTextStyle: TextStyle(
        fontSize: 18,
        color: Colors.green,
      ),
      showHeader: true,
      headerTextStyle: const TextStyle(color: Colors.pink),
      todayTextStyle: const TextStyle(
        color: Colors.blue,
      ),

      todayButtonColor: Colors.yellow,
      selectedDayTextStyle: const TextStyle(
        color: Colors.yellow,
      ),
      minSelectedDate: _currentDate.subtract(const Duration(days: 360)),
      maxSelectedDate: _currentDate.add(const Duration(days: 360)),
      prevDaysTextStyle: const TextStyle(
        fontSize: 16,
        color: Colors.pinkAccent,
      ),
      inactiveDaysTextStyle: const TextStyle(
        color: Colors.tealAccent,
        fontSize: 16,
      ),
      onCalendarChanged: (DateTime date) {
        setState(() {
          _targetDateTime = date;
          _currentMonth = DateFormat.yMMM().format(_targetDateTime);
        });
      },
      onDayLongPressed: (DateTime date) {},
    );

    super.initState();
  }

  Widget listTile() {
    return Column(
      children: [
        Container(
          margin: const EdgeInsets.symmetric(horizontal: 12),
          decoration: BoxDecoration(
              color: Colors.white60, borderRadius: BorderRadius.circular(12)),
          child: Row(
            crossAxisAlignment: CrossAxisAlignment.center,
            children: [
              Container(
                height: 50,
                width: 42,
                margin: const EdgeInsets.all(12),
                decoration: BoxDecoration(
                    shape: BoxShape.circle,
                    color: Colors.white,
                    border: Border.all(color: Colors.pink, width: 0.2)),
                child: const Center(
                  child: Text(
                    'HR',
                    style: TextStyle(color: Colors.pink),
                  ),
                ),
              ),
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: const [
                    Text(
                      'Home Rent',
                      style: TextStyle(
                          color: Colors.black, fontWeight: FontWeight.bold),
                    ),
                    SizedBox(
                      height: 4,
                    ),
                    Text('\$2000'),
                  ],
                ),
              ),
              const Icon(
                Icons.arrow_forward_ios_sharp,
                color: Colors.grey,
              ),
              const SizedBox(
                width: 12,
              )
            ],
          ),
        ),
        const SizedBox(
          height: 12,
        )
      ],
    );
  }

  Widget _getSearchField(context, onSearchText(text)) {
    return Container(
      width: MediaQuery.of(context).size.width,
      margin: const EdgeInsets.symmetric(horizontal: 16),
      decoration: BoxDecoration(
          borderRadius: BorderRadius.circular(12), color: Colors.white60),
      child: TextFormField(
        onChanged: onSearchText,
        decoration: const InputDecoration(
            border: InputBorder.none,
            hintText: '   Search...',
            suffixIcon: Icon(
              Icons.search,
              color: Colors.black,
            )),
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    return SafeArea(
      child: SingleChildScrollView(
        controller: _scrollController,
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          mainAxisAlignment: MainAxisAlignment.start,
          children: <Widget>[
            //custom icon

            Container(
              margin: const EdgeInsets.only(
                bottom: 16.0,
                left: 16.0,
                right: 16.0,
              ),
              child: Row(
                children: <Widget>[
                  const Expanded(
                      child: Text(
                    "Preview Expense",
                    style: TextStyle(
                      fontWeight: FontWeight.bold,
                      fontSize: 24.0,
                    ),
                  )),
                  IconButton(
                    icon: const Icon(Icons.payments_outlined),
                    onPressed: () {
                      setState(() {
                        _targetDateTime = DateTime(
                            _targetDateTime.year, _targetDateTime.month - 1);
                        _currentMonth =
                            DateFormat.yMMM().format(_targetDateTime);
                      });
                    },
                  ),
                ],
              ),
            ),
            Container(
              height: 400,
              margin: const EdgeInsets.symmetric(horizontal: 16.0),
              child: _calendarCarouselNoHeader,
            ),
            const SizedBox(
              height: 20,
            ),
            _getSearchField(context, (text) {
              _scrollController.animateTo(text.toString().isEmpty ? 0 : 430,
                  duration: const Duration(milliseconds: 500),
                  curve: Curves.ease);
            }),
            ListView.builder(
              padding: const EdgeInsets.only(top: 20),
              shrinkWrap: true,
              physics: const NeverScrollableScrollPhysics(),
              itemBuilder: (context, index) {
                return listTile();
              },
              itemCount: 10,
            )
          ],
        ),
      ),
    );
  }
}


lib/src/animated_hover_background_view.dart
library animated_hover_background;

import 'dart:ui';

import 'package:flutter/material.dart';
import 'package:rive/rive.dart';

class AnimatedHoverBackground extends StatelessWidget {
  Widget? widget_child;

   AnimatedHoverBackground({required Widget child,Key? key}) {
     this.widget_child=child;
   }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.white,
      body: Container(
        width: MediaQuery.of(context).size.width,
        height: MediaQuery.of(context).size.height,
        child: Stack(
          children: [
            RiveAnimation.asset("packages/animated_hover_background/anim/background_anim.riv",fit: BoxFit.fill,),
            ClipRRect(
              // borderRadius: BorderRadius.only(topLeft: Radius.circular(24),topRight: Radius.circular(24)),
              child: BackdropFilter(
                filter: ImageFilter.blur(
                  sigmaX: 100.0,
                  sigmaY: 100.0,
                ),
                child: Container(
                  width: MediaQuery.of(context).size.width,
                  height: MediaQuery.of(context).size.height,
                  child: widget_child,

                ),
              ),
            ),
          ],
        ),
      ),

    );
  }
}


example/lib/home_view.dart
import 'package:flutter/material.dart';

class Home_Dash_Screen extends StatefulWidget {
  const Home_Dash_Screen({Key? key}) : super(key: key);

  @override
  State<Home_Dash_Screen> createState() => _Home_Dash_ScreenState();
}

class _Home_Dash_ScreenState extends State<Home_Dash_Screen> {
  String pageName = 'home';

  Widget getExpenseText() {
    return Column(
      children: [
        const SizedBox(
          height: 20,
        ),
        Text(
          "This month spend",
          style: TextStyle(
              color: Colors.grey.shade400,
              fontSize: 16,
              fontWeight: FontWeight.bold),
        ),
        const SizedBox(
          height: 20,
        ),
        const Text(
          "\$1,200.11",
          style: TextStyle(
              color: Colors.black, fontSize: 32, fontWeight: FontWeight.w600),
        ),
        const SizedBox(
          height: 50,
        ),
      ],
    );
  }

  Widget getDashBoardTicket(BuildContext context) {
    return Container(
        margin: const EdgeInsets.only(left: 20, right: 20),
        height: 220,
        decoration: BoxDecoration(
            borderRadius: BorderRadius.circular(12),
            color: Colors.white.withOpacity(0.4),
            boxShadow: const [
              BoxShadow(
                  color: Colors.black12, offset: Offset(0, 0), blurRadius: 32)
            ]),
        child: Stack(
          children: [
            Align(
                alignment: Alignment.centerRight,
                child: Container(
                  margin: const EdgeInsets.only(right: 32),
                  child: Column(
                    mainAxisAlignment: MainAxisAlignment.center,
                    crossAxisAlignment: CrossAxisAlignment.center,
                    children: const [
                      Text(
                        "View More",
                        maxLines: 1,
                        style: TextStyle(
                          fontSize: 12,
                          fontWeight: FontWeight.bold,
                        ),
                      ),
                      SizedBox(
                        height: 12,
                      ),
                      Icon(Icons.arrow_forward_ios)
                    ],
                  ),
                )),
            Align(
                alignment: Alignment.centerLeft,
                child: Image.asset("assets/sales.png"))
          ],
        ));
  }

  AppBar _getAppbar(context, onSearchText(text)) {
    return AppBar(
        elevation: 0.0,
        backgroundColor: Colors.transparent,
        title: Container(
          width: MediaQuery.of(context).size.width,
          decoration: BoxDecoration(
              borderRadius: BorderRadius.circular(12), color: Colors.white24),
          child: TextFormField(
            onChanged: onSearchText,
            decoration: const InputDecoration(
                border: InputBorder.none,
                hintText: '   Search...',
                suffixIcon: Icon(
                  Icons.search,
                  color: Colors.black,
                )),
          ),
        ));
  }

  Widget listTile() {
    return Column(
      children: [
        Container(
          margin: const EdgeInsets.symmetric(horizontal: 12),
          decoration: BoxDecoration(
              color: Colors.white60, borderRadius: BorderRadius.circular(12)),
          child: Row(
            crossAxisAlignment: CrossAxisAlignment.center,
            children: [
              Container(
                height: 50,
                width: 42,
                margin: const EdgeInsets.all(12),
                decoration: BoxDecoration(
                    shape: BoxShape.circle,
                    color: Colors.white,
                    border: Border.all(color: Colors.pink, width: 0.2)),
                child: const Center(
                  child: Text(
                    'HR',
                    style: TextStyle(color: Colors.pink),
                  ),
                ),
              ),
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: const [
                    Text(
                      'Home Rent',
                      style: TextStyle(
                          color: Colors.black, fontWeight: FontWeight.bold),
                    ),
                    SizedBox(
                      height: 4,
                    ),
                    Text('\$2000'),
                  ],
                ),
              ),
              const Icon(
                Icons.arrow_forward_ios_sharp,
                color: Colors.grey,
              ),
              const SizedBox(
                width: 12,
              )
            ],
          ),
        ),
        const SizedBox(
          height: 12,
        )
      ],
    );
  }

  final ScrollController _scrollController = ScrollController();

  @override
  Widget build(BuildContext context) {
    return SingleChildScrollView(
      controller: _scrollController,
      child: Column(
        children: [
          _getAppbar(context, (text) {
            _scrollController.animateTo(430,
                duration: const Duration(milliseconds: 500),
                curve: Curves.ease);
          }),
          getExpenseText(),
          getDashBoardTicket(context),
          const SizedBox(
            height: 20,
          ),
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            crossAxisAlignment: CrossAxisAlignment.end,
            children: const [
              Padding(
                padding: EdgeInsets.only(left: 18.0),
                child: Text('Upcoming Plans'),
              ),
              Padding(
                padding: EdgeInsets.only(right: 18.0),
                child: Icon(Icons.sort_rounded),
              )
            ],
          ),
          ListView.builder(
            padding: const EdgeInsets.only(top: 20),
            shrinkWrap: true,
            physics: const NeverScrollableScrollPhysics(),
            itemBuilder: (context, index) {
              return listTile();
            },
            itemCount: 10,
          )
        ],
      ),
    );
  }
}
